EIPS/eip-6997.md

```diff
@@ -0,0 +1,371 @@
+        // The approved address.
+        address approve;
+        // The token Id.
+        uint256 tokenId;
+        // Wether is a total approvement.
+        bool approveAll;
+        // Whether is a valid approve.
+        bool valid;
+    }
+
+    /**
+     * @dev Emitted when a new transfer validation has been requested.
+     */
+    event ValidateTransfer(address indexed from, address to, uint256 indexed tokenId, uint256 indexed transferValidationId);
+
+    /**
+    * @dev Emitted when a new approve validation has been requested.
+    */
+    event ValidateApprove(address indexed owner, address approve, uint256 tokenId, bool indexed approveAll, uint256 indexed approveValidationId);
+
+    /**
+     * @dev Returns true if this contract is a validator ERC721.
+     */
+    function isValidatorContract() external view returns (bool);
+
+    /**
+     * @dev Returns the transfer validation struct using the transfer ID.
+     *
+     */
+    function transferValidation(uint256 transferId) external view returns (TransferValidation memory);
+
+    /**
+    * @dev Returns the approve validation struct using the approve ID.
+    *
+    */
+    function approveValidation(uint256 approveId) external view returns (ApproveValidation memory);
+
+    /**
+     * @dev Return the total amount of transfer validations created.
+     *
+     */
+    function totalTransferValidations() external view returns (uint256);
+
+    /**
+     * @dev Return the total amount of transfer validations created.
+     *
+     */
+    function totalApproveValidations() external view returns (uint256);
+}
+  ```
+
+The `isValidatorContract()` function MUST be implemented as `public`.
+
+The `transferValidation(uint256 transferId)` function MAY be implemented as `public` or `external`.
+
+The `approveValidation(uint256 approveId)` function MAY be implemented as `public` or `external`.
+
+The `totalTransferValidations()` function MAY be implemented as `pure` or `view`.
+
+The `totalApproveValidations()` function MAY be implemented as `pure` or `view`.
+
+## Rationale
+
+### Universality
+
+The standard only defines the validation functions, but not how they should be used.
+
+Like the [ERC-721](./eip-721.md) standard with the `_mint` and `_burn` functions, it defines the validations as internal and lets the user decide how to manage them.
+
+An example could be to have an address validator connected to a dApp so that users could manage their validations.
+
+This validator could be used for all NFTs or only for some users.
+
+It could also be used as an wrapped Smart Contract for existing ERC-721, allowing 1/1 conversion with existing NFTs.
+
+### Extensibility
+
+This standard only defines the validation function, but does not define the system with which it has to be validated. A third-party protocol can define how it wants to call these functions as it wishes.
+
+## Backwards Compatibility
+
+This standard is fully [ERC-721](./eip-721.md) compatible.
```

Comment:
I seriously doubt that this is fully backwards compatible. Please revisit this section after defining what "frozen" means above.

Reply:
The ERC-6997 adapt the ERC-721 to have an extra validation step, it has all the ERC-721 functions and only override `_transfer`, `_approve` and `_setApprovalForAll`.

I guess if it override something of the ERC-721 we can't consider it's fully backwards compatible? Still if it has the same functions with same signature.

I misunderstood the backwards compatibility, I can change it if you want 👍

Reply:
You want to consider more than just the Solidity interface when you're writing the backwards compatibility section. The way you have "frozen" defined right now would likely not be backwards compatible with ERC-721 contracts: for example, if `transferFrom` returns successfully, they'd expect the token to be transferred.

Reply:
Changed